### ISSUE TITLE
Setting row edge color of matplotlib table

### ISSUE THREAD

**Issue by abbazs** [matplotlib/matplotlib#12389]
I've a pandas DataFrame plotted as a table using matplotlib: [(from this answer)](https://stackoverflow.com/a/39358752/1437877)

Now I want to set the bottom edge color of a given row and I've this code:

**Code for reproduction**

```python
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib
import six

df = pd.DataFrame()
df['date'] = ['2016-04-01', '2016-04-02', '2016-04-03', '2016-04-04']
df['calories'] = [2200, 2100, 1500, 1800]
df['sleep hours'] = [2200, 2100, 1500, 1500]
df['gym'] = [True, False, False, True]

def render_mpl_table(data, col_width=3.0, row_height=0.625, font_size=14,
                     header_color='#40466e', row_colors=['#f1f1f2', 'w'], edge_color='w',
                     bbox=[0, 0, 1, 1], header_columns=0,
                     ax=None, **kwargs):
    if ax is None:
        size = (np.array(data.shape[::-1]) + np.array([0, 1])) * np.array([col_width, row_height])
        fig, ax = plt.subplots(figsize=size)
        ax.axis('off')

    mpl_table = ax.table(cellText=data.values, bbox=bbox, colLabels=data.columns, **kwargs)

    mpl_table.auto_set_font_size(False)
    mpl_table.set_fontsize(font_size)

    for k, cell in six.iteritems(mpl_table._cells):
        cell.set_edgecolor(edge_color)
        if k[0] == 0 or k[1] < header_columns:
            cell.set_text_props(weight='bold', color='w')
            cell.set_facecolor(header_color)
        else:
            cell.set_facecolor(row_colors[k[0]%len(row_colors) ])
    return ax

def get_table(ax):
    table = None
    for child in ax.get_children():
        if isinstance(child, matplotlib.table.Table):
            table = child
            return table
    return table

def set_row_edge_color(ax, row, color):
    table = get_table(ax)
    for k, cell in  six.iteritems(table._cells):
        if (k[0] == row):
            cell.set_edgecolor(color)

ax = render_mpl_table(df, header_columns=0, col_width=2.0)
set_row_edge_color(ax, 2, 'k')
plt.show()
```

I'm unable to set only the color of row bottom and it gets set like this: 
![yxkdd 1](https://user-images.githubusercontent.com/758268/46427503-3a602480-c75f-11e8-9d64-eea95030ba4f.png)

Is there a way to set only the row bottom color like this?

![n3hdv 1](https://user-images.githubusercontent.com/758268/46427538-54016c00-c75f-11e8-8cb7-303bed739fea.png)

Or is there a way to locate row in the figure/plot and draw a horizontal line?

**Comment by timhoffm** [comment]
The table cells are not that sophisticated that they support different edge colors on different edges.

For drawing a line, you can use the following workaround

~~~
def set_row_edge_color(ax, row, color):
    table = get_table(ax)
    y = None
    for k, cell in  six.iteritems(table._cells):
        if k[0] == row:
            y = cell.xy[1]
            ax.axhline(y, color=color, linewidth=4)
            return

ax = render_mpl_table(df, header_columns=0, col_width=2.0)
plt.draw()
set_row_edge_color(ax, 2, 'k')
plt.show()
~~~

Note that you have to call `plt.draw()` before calling `set_row_edge_color()`. Otherwise, the cells are not yet laid out in the figure and `cell.xy` is not yet the correct position.

**Comment by abbazs** [comment]
I'm sorry - it doesn't solve the issue. As the cell.xy[1] returns the y of the last row and not the row required!
For all the rows cell.xy[1] always returns 0!

**Comment by ImportanceOfBeingErnest** [comment]
Note that the issue tracker is meant to be used for issues with the library. There does not seem to be any bug involved here. Closing again.

You did the right thing by asking [your question](https://stackoverflow.com/questions/52632356/setting-row-edge-color-of-matplotlib-table) on stackoverflow. Just wait a bit for an answer to come up. I just answered [your question](https://stackoverflow.com/questions/52632356/setting-row-edge-color-of-matplotlib-table) over there. Feel free to comment if that is not what you're after.
